src/gui/src/layoutViewer.cpp

```diff
@@ -2203,11 +2231,52 @@ void LayoutViewer::drawPinMarkers(Painter& painter,
             text_anchor_pt.setY(text_anchor_pt.y() - max_dim - text_margin);
           }
 
-          painter.drawString(text_anchor_pt.x(), text_anchor_pt.y(), text_anchor, term->getName());
+          PinText pin_specs;
+          pin_specs.layer = layer;
+          pin_specs.text = term->getName();
+          pin_specs.pt = text_anchor_pt;
+          pin_specs.anchor = text_anchor;
+          pin_specs.can_rotate = arg_min == 2 || arg_min == 3;
+          // only need bounding box when rotation is possible
+          if (pin_specs.can_rotate) {
+            odb::Rect text_rect = painter.stringBoundaries(pin_specs.pt.x(), pin_specs.pt.y(), pin_specs.anchor, pin_specs.text);
+            text_rect.bloat(text_margin, text_rect);
+            pin_specs.rect = Search::Box(Search::Point(text_rect.xMin(), text_rect.yMin()),
+                                         Search::Point(text_rect.xMax(), text_rect.yMax()));
+            pin_text_spec_shapes[layer].insert(pin_specs.rect);
```

Comment:
If you have overlapping pin text on different layers you should still rotate.

I find that avoiding rotating only works if you are zoomed very far in.  Is this worth the trouble?